### PR TITLE
Fix - Avoid usage of JsonConvert in test host process

### PR DESCRIPTION
**Pull request by abhishkk**
## Related issue
https://github.com/Microsoft/vstest/issues/1757

### CODE REVIEW

File: src/Microsoft.TestPlatform.CommunicationUtilities/JsonDataSerializer.cs [hunk 1]
```diff
@@ -36,6 +37,7 @@ private JsonDataSerializer()
                 ReferenceLoopHandling = ReferenceLoopHandling.Ignore
             };
 
+            messageSerializer = JsonSerializer.Create();
```
Comment (ObeA) [inline src/Microsoft.TestPlatform.CommunicationUtilities/JsonDataSerializer.cs]: Did you test if this solves #1757 ? According to the [docs ](https://www.newtonsoft.com/json/help/html/T_Newtonsoft_Json_JsonSerializer.htm) .Create() uses the default settings, which I think caused the issue.

Comment (abhishkk) [reply src/Microsoft.TestPlatform.CommunicationUtilities/JsonDataSerializer.cs]: @ObeA As mentioned by @singhsarab , `Create()` doesn't use default settings of JsonConvert. Check [docs](https://www.newtonsoft.com/json/help/html/M_Newtonsoft_Json_JsonSerializer_Create.htm) here.

**Comment by abhishkk** [comment]
@dotnet-bot test this please

**Review by cltshivash — APPROVED** [review]
:shipit:

File: test/Microsoft.TestPlatform.CommunicationUtilities.UnitTests/JsonDataSerializerTests.cs [hunk 2]
```diff
@@ -22,6 +24,40 @@ public JsonDataSerializerTests()
...
+            {
+                ContractResolver = new DefaultContractResolver
+                {
+                    NamingStrategy = new SnakeCaseNamingStrategy()
+                }
+            };
+
+            var classWithSelfReferencingLoop = new ClassWithSelfReferencingLoop(null);
+            classWithSelfReferencingLoop = new ClassWithSelfReferencingLoop(classWithSelfReferencingLoop);
+            classWithSelfReferencingLoop.InfiniteRefernce.InfiniteRefernce = classWithSelfReferencingLoop;
```
Comment (singhsarab) [inline test/Microsoft.TestPlatform.CommunicationUtilities.UnitTests/JsonDataSerializerTests.cs]: Nit: InfiniteReference*

**Review by singhsarab — APPROVED** [review]

**Pull request merged**